lib/board.rb

```diff
@@ -0,0 +1,65 @@
+class Board
+  attr_accessor :cells, :turn_count, :exes, :oes
+
+  def initialize
+    @cells = Array.new(9, " ")
+    @exes = []
+    @oes = []
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(input)
+    index = input.to_i - 1
+    @cells[index]
+  end
+
+  def exes
+    @exes = cells.map.with_index.map { |mark, i| i if mark == "X" }
+  end
+
+  def oes
+    @oes = cells.map.with_index.map { |mark, i| i if mark == "O" }
+  end
+
+  def full?
+    return false if @cells.any? { |e| e == " " }
+    true
+  end
+
+  def taken?(pos)
+    self.position(pos) != " "
+  end
+
+  def turn_count
+    turn_count = 0
+    @cells.each { |e| turn_count += 1 if e != " " }
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```